src/Npgsql/TypeHandlers/NumericHandlers/DecimalRaw.cs

```diff
@@ -51,7 +53,7 @@ public int Scale
         public uint Mid => _mid;
         public uint Low => _low;
 
-        public DecimalRaw(long value)
+        public DecimalRaw(long value) : this()
```

Comment:
Would it hurt performance if it sets 16 bytes twice?

Reply:
I'm not sure what you mean 🤔

Reply:
If I understand correctly, `this` is to zero initialize memory using the default `ctor`. After that the fields are set using the provided value. It looks like setting fields twice.

Reply:
AFAIK, I don't think it's setting any fields twice; https://sharplab.io/#v2:C4LglgNgPgAgDAAhgRgNwFgBQ8nIHQBKArgHbBgC2ApngJJlUBOA9gA4DKTAbmAMZUBnDJiwBtdsEZFewADIBDAJ7MiwABQLlqgNJgSAEzwBRAB6sIfMMACUAXSwDJ04AgDiRMPoLyA7lgDeWAjBCKIAYmBUEPoA8gBmcQJU6nBwdkgAzG4e+ggAavIQRFTCIaERUbEJSSlptpkIei4AIvLA8mhBIeGR0fGJyWpwACzpMFkCABbMjC1t8gBMpd0VfdWDcABsYxPTswit7RnLwT2V/TVDABw7CBDMJADmB/PDJw3unt4+ap+5XIVitYECAEMBJmABGpgQBeAB8+UBVAQMIQAKKJSwAF8gA===

Reply:
I meant that it initializes the memory and then sets the fields, but looking at the benchmark results everything is fine.